metrics/network/network-metrics-iperf3.sh

```diff
@@ -0,0 +1,540 @@
+	read -a server_results <<< ${server_result}
+	local total_bidirectional_client_bandwidth=${client_results[6]}
+	local total_bidirectional_client_bandwidth_units=${client_results[7]}
+	local total_bidirectional_server_bandwidth=${server_results[6]}
+	local total_bidirectional_server_bandwidth_units=${server_results[7]}
+
+	local json="$(cat << EOF
+	{
+		"network iperf3 bidir bw client to server": {
+			"Result" : $total_bidirectional_client_bandwidth,
+			"Units"  : "$total_bidirectional_client_bandwidth_units"
+		},
+		"network iperf3 bidir bw server to client": {
+ 			"Result" : $total_bidirectional_server_bandwidth,
+			"Units"  : "$total_bidirectional_server_bandwidth_units"
+		}
+	}
+EOF
+)"
+
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Results"
+	metrics_json_save
+	clean_env
+}
+
+# This function checks/verify if the iperf3 server
+# is ready/up for requests.
+function check_iperf_server() {
+	# timeout of 3 seconds aprox.
+	local time_out=6
+	local period=0.5
+	local test_cmd="iperf3 -c "$server_address" -t 1"
+
+	# check tools dependencies
+	local cmds=("netstat")
+	check_cmds "${cmds[@]}"
+
+	while [ 1 ]; do
+		if ! bash -c "$test_cmd" > /dev/null 2>&1; then
+			echo "waiting for server..."
+			count=$((count+1))
+			sleep $period
+		else
+			echo "iperf server is up!"
+			break;
+		fi
+
+		if [ "$count" == "$time_out" ]; then
+			die "iperf server init fails"
+		fi
+	done
+
+	# Check listening port
+	lport="$(netstat -atun | grep "$port" | grep "LISTEN")"
+	if [ -z "$lport" ]; then
+		die "port is not listening"
+	fi
+}
+
+# This function parses the output of iperf execution
+function parse_iperf_bwd() {
+	local TEST_NAME="$1"
+	local result="$2"
+
+	metrics_json_init
+
+	if [ -z "$result" ]; then
+		die "no result output"
+	fi
+
+	# Save configuration
+	metrics_json_start_array
+
+	# Filter receiver/sender results
+	rx_res=$(echo "$result" | grep "receiver" | awk -F "]" '{print $2}')
+	tx_res=$(echo "$result" | grep "sender" | awk -F "]" '{print $2}')
+
+ 	# Getting results
+	rx_bwd=$(echo "$rx_res" | awk '{print $5}')
+	tx_bwd=$(echo "$tx_res" | awk '{print $5}')
+	rx_uts=$(echo "$rx_res" | awk '{print $6}')
+	tx_uts=$(echo "$tx_res" | awk '{print $6}')
+
+	local json="$(cat << EOF
+	{
+ 		"${TEST_NAME} receiver": {
+			"Result" : $rx_bwd,
+			"Units"  : "$rx_uts"
+		},
+		"${TEST_NAME} sender": {
+			"Result" : $tx_bwd,
+			"Units"  : "$tx_uts"
+		}
+	}
+EOF
+)"
+
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Results"
+	metrics_json_save
+}
+
+# This function parses the output of iperf UDP execution, and
+# saves the receiver successful datagram value in the results.
+function parse_iperf_pps() {
+	local TEST_NAME="$1"
+	local result="$2"
+
+	metrics_json_init
+
+	if [ -z "$result" ]; then
+		die "no result output"
+	fi
+
+	# Save configuration
+	metrics_json_start_array
+
+	# Extract results
+	local lost=$(echo "$result" | jq '.end.sum.lost_packets')
```

Comment:
Given the input we get here is JSON - we can I believe dump that whole input into the JSON results file as well as the data we cook out of it.